### PR TITLE
Adding some orphaned tests back.

### PR DESCRIPTION
**Pull request by VSadov**
There was a recent discovery of two files containing ref locals/return tests that were not linked to a project.
The tests seems to be from pre-C#6 experimental times and many have the counterparts in the current tests.

There is a subset of compile error tests for which the overlapping coverage is not so clear so I am porting/adding those.

### CODE REVIEW

**Comment by VSadov** [comment]
@dotnet/roslyn-compiler - FYI test only change. Salvaging some old ref locals/returns tests.

File: src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs [hunk 1]
```diff
@@ -1385,5 +1385,893 @@ void Write(ref int y)
...
+
+    ref D M()
+    {
+        return ref d;
+    }
+}
+";
+
+            CreateCompilationWithMscorlib46(text).VerifyDiagnostics(
+            );
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs]: Should this report errors or should the test name be changed?

Comment (VSadov) [inline src/Compilers/CSharp/Test/Semantic/Semantics/RefLocalsAndReturnsTests.cs]: Looks like a bug actually. Should be an error. Thanks!!!

**Comment by cston** [comment]
LGTM

**Comment by VSadov** [comment]
test windows_debug_vs-integration_prtest please

**Comment by VSadov** [comment]
Roslyn.VisualStudio.IntegrationTests.VisualBasic.BasicGoToDefinition.ObjectBrowserNavigation [FAIL]

**Comment by VSadov** [comment]
The change has nothing to do with VB

**Comment by VSadov** [comment]
test windows_debug_vs-integration_prtest please

**Pull request merged**